raiden/transfer/routes.py

```diff
@@ -40,14 +47,28 @@ def filter_acceptable_routes(
 
 
 def prune_route_table(
-    route_states: List[RouteState], selected_route: RouteState
+    route_states: List[RouteState],
+    selected_route: RouteState,
+    initiator_address: InitiatorAddress,
 ) -> List[RouteState]:
     """Given a selected route, returns a filtered route table that
     contains only routes using the same forward channel and removes our own
     address in the process.
+    Our address is also removed from the address metadata, if we are not the initiator
```

Comment:
Currently, it would be easier for the receiver note if this would not happen, and we would include our (the senders) address-metadata all the time.
Otherwise this transport would have to inject the user-id (so that this can in turn be memorised in the state 
for later creation of send-events that go back to the sender).

Reply:
Above described behaviour has been implemented